tests/php/RecursiveStagesServiceTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace SilverStripe\Versioned\Tests;
+
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Versioned\Tests\RecursiveStagesServiceTest\ChildObject;
+use SilverStripe\Versioned\Tests\RecursiveStagesServiceTest\ColumnObject;
+use SilverStripe\Versioned\Tests\RecursiveStagesServiceTest\GroupObject;
+use SilverStripe\Versioned\Tests\RecursiveStagesServiceTest\PrimaryObject;
+use SilverStripe\Versioned\Versioned;
+
+class RecursiveStagesServiceTest extends SapphireTest
+{
+    /**
+     * @var string
+     */
+    protected static $fixture_file = 'RecursiveStagesServiceTest.yml';
+
+    /**
+     * @var string[]
+     */
+    protected static $extra_dataobjects = [
+        PrimaryObject::class,
+        ColumnObject::class,
+        GroupObject::class,
+        ChildObject::class,
+    ];
+
+    protected static $required_extensions = [
+        PrimaryObject::class => [
+            Versioned::class,
+        ],
+        GroupObject::class => [
+            Versioned::class,
+        ],
+        ChildObject::class => [
+            Versioned::class,
+        ],
+    ];
+
+    public function testStageDiffersRecursiveWithInvalidObject(): void
+    {
+        Versioned::withVersionedMode(function (): void {
+            Versioned::set_stage(Versioned::DRAFT);
+
+            /** @var PrimaryObject|Versioned $primaryItem */
+            $primaryItem = PrimaryObject::create();
```

Comment:
```suggestion
            $primaryItem = PrimaryObject::create();
            $primaryItem->Title = 'This registers as a change in DataObject::isChanged()';
```
Just to make sure - otherwise we could theoretically be seeing no changes just because there _were_ no changes, rather than because we're intentionally not checking it.

Reply:
All good - I've added your change. Feel free to merge once tests pass.